app/controllers/donations_controller.rb

```diff
@@ -84,7 +84,7 @@ def update
     redirect_to donations_path
   rescue => e
     flash[:alert] = "Error updating donation: #{e.message}"
-    render "edit"
+    redirect_back(fallback_location: edit_donation_path)
```

Comment:
Wouldn't this lose all the information you've entered?

Reply:
That is the existing behavior before this bug fix, I can try to change it and add a test so that it keeps all entered information except for the incorrect edited quantity if you want?

Reply:
That seems to be the correct thing to do, yes.

Reply:
I ended up keeping all entered information including the incorrect edited quantity in commit [a393e13]( https://github.com/rubyforgood/human-essentials/pull/4483/commits/a393e1372acf682806e5fcd40227bee6fabadeeb), if that doesn't work let me know.

Note that per #4508 when testing READ_EVENTS must be set to true before a fixed donation can be submitted after an error.